### DISCUSSION TITLE
Unable to add memory limit to functions

### DISCUSSION THREAD

**Discussion by akshatmittal** [vercel/vercel#4100] [General] [answered]
I'm unable to add memory limit to the functions. I'm trying to add a memory limit of 256MB, but the Functions Deployment still tells me the memory size is 1GB.

My `now.json` files looks like so:

```json
{
  "functions": {
    "src/pages/api/*": {
      "memory": 256,
      "maxDuration": 10
    }
  }
}
```

**Comment by styfle** [comment]
I think you need too remove `src`. Also the single `*` only matches one level deep. Try this:

```json
{
  "functions": {
    "pages/api/**/*.js": {
      "memory": 256
    }
  }
}
```

If you are using `getServerSideProps()` for any pages, you'll need to include all, not just API.

```json
{
  "functions": {
    "pages/**/*": {
      "memory": 256
    }
  }
}

```

**Comment by akshatmittal** [comment, marked as answer]
Hey @styfle, thanks for the answer. I tried a couple of different configurations with that, and it seems like I do need the `src` in there (since my code is in the src folder), it was just the double `**` that I was missing. My bad.

At the end, this worked for me:
```js
{
  "functions": {
    "src/pages/**/*": {
      "memory": 256,
      "maxDuration": 5
    }
  }
}
```

Thank you for your help!
